test/execution/server_test.exs

```diff
@@ -72,13 +86,13 @@ defmodule Wanda.Execution.ServerTest do
       assert ExecutionResult |> Repo.all() |> Enum.empty?()
     end
 
-    test "should exit when all facts are sent by all agents" do
+    test "should exit when all facts are sent by all agents", state do
```

Comment:
the "state" here is usually referred as "context". Also you can already pattern match %{check: %{id: id}}

Reply:
@fabriziosestito About removing the catalog fixtures completely: we wouldn't be testing the load from yaml. That's why they are still some there. They are not used for anything else. At the end, even though it is simple, we do some operations there that are good to test.

Reply:
@arbulu89 ah ok that make sense